docs/csharp/codedoc.md

```diff
@@ -2,26 +2,547 @@
+
+XML documentation tags are specially formatted comments, added above the definition of any user defined type or member, used to document your code.
+Unlike regular C# comments documentation comments are not ignored by the compiler and you can generate the XML documentation file at compile time
+by adding `"xmlDoc": true` under `buildOptions` in your `project.json` when using .NET Core or use the `/doc` compiler option for the .NET framework.
+Go [here](https://msdn.microsoft.com/en-us/library/3260k4x7.aspx) to learn how to enable XML documentation generation in Visual Studio.
+
+Documentation tags are commented using three forward slashes (`///`) as opposed to two forward slashes (`//`) used for single line comments.
+There are tools tools like [DocFX](https://dotnet.github.io/docfx/) and [Sandcastle](https://github.com/EWSoftware/SHFB) that help you generate documentation websites from your documentation tags,
+also, distributing the compiler generated XML file along with your library allows Visual Studio and other IDEs to show quick information about types or members when performing intellisense. 
+
+<a name="summary"></a>
+## &lt;summary&gt;
+
+The &lt;summary&gt; tag contains the primary information of a class or method.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+    /// <summary>
+    /// This method does stuff.
+    /// </summary>
+    public void DoStuff(int a)
+    {
+
+    }
+
+    /// <summary>
+    /// This method does more stuff.
+    /// </summary>
+    public void DoMoreStuff(int a, string b)
+    {
+
+    }
+}
+```
+
+<a name="remarks"></a>
+## &lt;remarks&gt;
+
+The &lt;remarks&gt; tag is used to add information about a type or type members, 
+supplementing the information specified with &lt;summary&gt;.
+
+```csharp
+/// <summary>
+/// This class primarily does something.
+/// </summary>
+/// <remarks>
+/// This class can also do everything.
+/// </remarks>
+public class SomeClass
+{
+    
+}
+```
+
+<a name="returns"></a>
+## &lt;returns&gt;
+
+The &lt;returns&gt; tag should be used in the comment for a method declaration to describe the return value.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+    /// <summary>
+    /// This method does stuff.
+    /// </summary>
+    /// <returns>Stuff it has done.</returns>
+    public int DoStuff(int a)
+    {
+        return a;
+    }
+}
+```
+
+<a name="example"></a>
+## &lt;example&gt;
+
+The &lt;example&gt; tag lets you specify an example of how to use a method or other library member. 
+This commonly involves using the [&lt;code&gt;](#code) and sometimes the [&lt;c&gt;](#c) tag.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+    /// <summary>
+    /// This method does stuff.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// SomeClass someClass = new SomeClass();
+    /// someClass.DoStuff();
```

Comment:
I'd really like to have an example here that is long enough to show some indenting. That would make it clear to readers that examples are formatted such that whitespace is significant.